src/app/app.component.ts

```diff
@@ -1,10 +1,60 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { extend } from 'webdriver-js-extender';
+
+class Point {
+  x: number; y: number;
+    constructor(x: number, y: number) {
+      this.x = x;
+      this.y = y;
+  }
+
+  add(newPoint: Point) {
+    return newPoint = new Point(newPoint.x + this.x, newPoint.y + this.y);
+  }
+}
+
+class Point3D extends Point {
+  x: number;
+  y: number;
```

Comment:
if we are extending Point, do we need to define x & y fields in this class?

Reply:
yes since we are getting them from the base class

Reply:
we don't need to define them in the derived class since they are already part of the base class.